GVFS/GVFS.Common/GVFSEnlistmentStatistics.cs

```diff
@@ -0,0 +1,189 @@
+﻿using GVFS.Common.Database;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace GVFS.Common
+{
+    public class GVFSEnlistmentStatistics
+    {
+        private List<string> gitPaths;
+        private List<string> placeholderFolderPaths;
+        private List<string> placeholderFilePaths;
+        private List<string> modifiedFolderPaths;
+        private List<string> modifiedFilePaths;
+        private int trackedFilesCount;
+        private int placeholderCount;
+        private int modifiedPathsCount;
+        private Dictionary<string, int> trackedFilesDirectoryTracking;
+        private Dictionary<string, int> hydratedFilesDirectoryTracking;
+
+        public GVFSEnlistmentStatistics(List<string> gitPaths, List<string> placeholderFolderPaths, List<string> placeholderFilePaths, List<string> modifiedFolderPaths, List<string> modifiedFilePaths)

```

Comment:
I think it'd be quite easy with this class. The way its designed is to use the constructor once for the expensive operations (IE using the git index to read and parse all files, or talking across the pipe, or going to the data structure underneath) which are client side, but then not ask them to do the expensive stuff over and over again and rather just take a focus into the calculation. Consider this use case with a GUI

1) User runs utility on entire enlistment, views data about most hydrated directories
2) User then wants to look closer at one directory in particular, so they click on it
3) Underneath, you only have to call `CalculateStatistics(clickedDirectory)` to get all of that information

I think that this way of doing it will reduce overhead and make interactivity simpler.